### ISSUE TITLE
Lint/NumberConversion doesn't handle `to_i` in symbol form

### ISSUE THREAD

**Issue by satyanash** [rubocop-hq/rubocop#7902]
Running `rubocop --only "Lint/NumberConverion" --auto-correct` does not handle the following cases:

```
"1,2,3,foo,5,6,7,8".split(',').map(&:to_i)

"foo".try(:to_i) # when using Rails, returns a 0
```

--------

## Expected behavior

At the minimum, **it should report a violation**.

For auto-correction, I'm not completely sure if this will work for all scenarios, but it could look like the following:
```
"1,2,3,foo,5,6,7,8".split(',').map { |i| Integer(i) } 
```
and
```
"foo".try{ |n| Integer(n) }
```

## Actual behavior
The above violations of using `to_i` are not reported.

## Steps to reproduce the problem

Just use any of the above examples in a file and run rubocop over them.

## RuboCop version
```
$ rubocop -V
0.78.0 (using Parser 2.6.5.0, running on jruby 2.5.3 java)
```

**Comment by rrosenblum** [comment]
This makes sense to me. The auto-corrections that you suggested looks accurate.
